sources/packages/backend/apps/api/src/route-controllers/designation-agreement/models/designation-agreement.dto.ts

```diff
@@ -1,49 +1,136 @@
  */
-export class UpdateDesignationAPIInDto {
+export class UpdateDesignationAPIInDTO {
   @IsEnum(DesignationAgreementStatus)
   designationStatus: DesignationAgreementStatus;
   @ValidateIf(
-    (value: UpdateDesignationAPIInDto) =>
+    (value: UpdateDesignationAPIInDTO) =>
       value.designationStatus === DesignationAgreementStatus.Approved,
   )
   @IsNotEmpty()
   @IsDateString()
   startDate?: string;
   @ValidateIf(
-    (value: UpdateDesignationAPIInDto) =>
+    (value: UpdateDesignationAPIInDTO) =>
       value.designationStatus === DesignationAgreementStatus.Approved,
   )
   @IsNotEmpty()
   @IsDateString()
   endDate?: string;
   @ValidateIf(
-    (value: UpdateDesignationAPIInDto) =>
+    (value: UpdateDesignationAPIInDTO) =>
       value.designationStatus === DesignationAgreementStatus.Approved,
   )
   @IsNotEmpty()
   @IsArray()
   @ValidateNested({ each: true })
-  @Type(() => UpdateDesignationLocation)
-  locationsDesignations?: UpdateDesignationLocation[];
+  @Type(() => DesignationLocationAPIInDTO)
+  locationsDesignations?: DesignationLocationAPIInDTO[];
+  @IsNotEmpty()
+  @MaxLength(NOTE_DESCRIPTION_MAX_LENGTH)
+  note: string;
+}
+
+/**
+ * This DTO contains dynamic data that must
+ * be validated by the form.io dryrun validation.
+ */
+export class SubmitDesignationAgreementAPIInDTO {
+  @Allow()
+  dynamicData: any;
+  @Allow()
+  locations: SubmittedLocationsAPIInDTO[];
+}
+
+/**
+ * This DTO contains dynamic data that must
+ * be validated by the form.io dryrun validation.
+ */
+export class SubmittedLocationsAPIInDTO {
+  @Allow()
+  locationId: number;
+  @Allow()
+  requestForDesignation: boolean;
+}
+
+export class DesignationAgreementAPIOutDTO {
+  designationId: number;
+  designationStatus: DesignationAgreementStatus;
+  locationsDesignations: LocationDesignationAPIOutDTO[];
+  submittedData: any;
+  startDate: string;
+  endDate: string;
+  institutionId: number;
+  institutionName: string;
+  institutionType: string;
+  isBCPrivate: boolean;
+}
+
+export class LocationDesignationAPIOutDTO {
+  designationLocationId?: number;
+  locationId: number;
+  locationName: string;
+  locationData: InstitutionLocationData;
+  requested: boolean;
+  approved?: boolean;
+}
+
+export class DesignationAgreementDetailsAPIOutDTO {
+  designationId: number;
+  designationStatus: DesignationAgreementStatus;
+  submittedDate: Date;
+  startDate?: string;
+  endDate?: string;
+}
+
+export class PendingDesignationAgreementDetailsAPIOutDTO extends DesignationAgreementDetailsAPIOutDTO {
+  legalOperatingName: string;
+}
+
+export class DesignationLocationAPIInDTO {
+  @Min(1)
+  locationId: number;
+  @IsBoolean()
+  approved: boolean;
+}
+
+/**
+ * startDate, endDate and locationsDesignations used only for approval.
+ */
+export class UpdateDesignationDetailsAPIInDTO {
+  @IsEnum(DesignationAgreementStatus)
+  designationStatus: DesignationAgreementStatus;
+  @IsNotEmpty()
+  @IsDateString()
+  startDate?: string;
```

Comment:
If `startDate` and `endDate` are decorated with `@IsNotEmpty()` so I am assuming that they are not nullable, right?

Reply:
Yes, I checked the use case that uses that DTO and requires both dates.

Reply:
Maybe the request was not clear but if the dates are required, can we change them to `startDate: string;` and `endDate: string;`?

Reply:
yes, you're right. Thanks!